api/src/pcapi/admin/custom_views/venue_view.py

```diff
@@ -246,49 +249,70 @@ def delete_model(self, venue: Venue) -> bool:
             flash("Impossible d'effacer un lieu pour lequel il existe des réservations.", "error")
         return False
 
-    def update_model(self, new_venue_form: Form, venue: Venue) -> bool:
-        has_siret_changed = new_venue_form.siret.data != venue.siret
-        if has_siret_changed:
-            bu_with_same_target_siret = finance_repository.find_business_unit_by_siret(new_venue_form.siret.data)
-            if bu_with_same_target_siret:
+    def update_model(self, edit_venue_form: Form, venue: Venue) -> bool:
+        update_siret = False
+        current_siret = venue.siret
+        new_siret = edit_venue_form.siret.data.strip() if edit_venue_form.siret.data else None
+        existing_pricing_point_id = None
+        unavailable_sirene = False
+        if current_siret and not new_siret:
+            flash("Le champ SIRET ne peut pas être vide si ce lieu avait déjà un SIRET.")
+            return False
+
+        if new_siret and new_siret != current_siret:
+            try:
+                finance_validation.validate_siret_format(new_siret)
+            except finance_exceptions.InvalidSiret:
+                flash("Le SIRET doit être une série d'exactement 14 chiffres.")
+                return False
+            existing_pricing_point_id = venue.current_pricing_point_id
+            if existing_pricing_point_id and venue.id != existing_pricing_point_id:
                 flash(
-                    f"Ce SIRET a déjà été attribué au point de remboursement {bu_with_same_target_siret.name},"
-                    f" il ne peut pas être attribué à ce lieu.",
-                    "error",
+                    f"Ce lieu a déjà un SIRET et un point de valorisation (id={existing_pricing_point_id}),"
+                    f" on ne peut pas modifier son SIRET"
```

Comment:
On est dans le bloc `if new_siret and new_siret != current_siret`, donc je pense qu'ici le lieu n'a pas forcément de SIRET. (D'ailleurs, si son point de valorisation n'est pas lui-même, le lieu ne devrait pas avoir de SIRET. Sinon il serait son propre point de valorisation). Donc je pense que le message d'erreur devrait plutôt être : 

```suggestion
                    f"Ce lieu a déjà un point de valorisation (id={existing_pricing_point_id}). Définir un SIRET "
                    f"pour ce lieu impliquerait qu'il soit son propre point de valorisation, donc de changer son "
                    f"point de valorisation, ce qui n'est pas autorisé."
```

La fin est peut-être un peu trop verbeuse (tu peux garder ta version si tu préfères), mais le début est à mon avis plus correct.

Reply:
Bien vu !
Je pars sur:
```suggestion
                    f"Ce lieu a déjà un SIRET et un point de valorisation (Venue.id={existing_pricing_point_id}). "
                    f"Définir un SIRET impliquerait qu'il soit son propre point de valorisation, "
                    f"mais le changement de point de valorisation n'est pas autorisé"
```

Reply:
Ta phrase commence par "Ce lieu a déjà un SIRET". Mais, justement (sauf si j'ai mal compris le code !), rien ne dit que le lieu a un SIRET, ici, ou bien ?

Reply:
Zut, c'est une typo, je corrige !